src/corehost/cli/json.h

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#ifndef __JSON_H__
+#define __JSON_H__
+
+#include "pal.h"
+#include "rapidjson/document.h"
+#include "rapidjson/fwd.h"
+
+class json_parser_t {
+    public:
+        ~json_parser_t() { delete[] m_json; }
+
+        const rapidjson::Document& document() const { return m_document; }
+
+        bool parse_file(const pal::string_t& path);
+        bool parse_string(const pal::string_t& str, const pal::string_t& context);
+
+    private:
+        pal::char_t *m_json{nullptr};
```

Comment:
~~pal::char_t *m_json{nullptr};~~ -> `std::unique_ptr<pal::char_t[]> m_json;`

Reply:
Then the destructor can be removed.

Reply:
> @lpereira Can you also provide the binary size comparison for both with Casablanca and with RapidJSON. The startup win is awesome!
> 

These are the sizes with RapidJSON (release mode, Linux, x86-64, stripped):
```
   text	   data	    bss	    dec	    hex	filename
 306034	   2292	   1056	 309382	  4b886	libhostfxr.so
 258763	   2612	    800	 262175	  4001f	libhostpolicy.so
```

And these are the sizes with Casablanca (release mode, Linux, x86-64, stripped):
```
   text	   data	    bss	    dec	    hex	filename
 340175	   4340	   1088	 345603	  54603	libhostfxr.so
 294478	   4636	    840	 299954	  493b2	libhostpolicy.so

```

Which is pretty nifty, too: ~69KB of text + ~9KB of data shaved, for ~79KB of gain in size.

Reply:
Old habits die hard. Fixed.